packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java

```diff
@@ -30,63 +30,44 @@ private SharePlugin(Registrar registrar) {
 
   @Override
   public void onMethodCall(MethodCall call, MethodChannel.Result result) {
-    if (call.method.equals("share")) {
-      if (!(call.arguments instanceof Map)) {
-        throw new IllegalArgumentException("Map argument expected");
-      }
-      // Android does not support showing the share sheet at a particular point on screen.
+    // Android does not support showing the share sheet at a particular point on screen.
+    if ("share".equals(call.method)) {
```

Comment:
I tried out a switch-case, but since there's only two options I think a simple if-elseif is better.

Reply:
OK.